### ISSUE TITLE
An error occurred while creating DB tree index: object 'Polyline' has no parent

### ISSUE THREAD

**Issue by dgirardeau** [CloudCompare/CloudCompare#489]
Random bug in the Interactive Segmentation tool:
See http://www.danielgm.net/cc/forum/viewtopic.php?t=2188

**Comment by luca-penasa** [comment]
The same happens when using the polyline tracing tool. 
In the specific case this happens when no cloud is selected before starting the tracing tool. 
If a cloud is selected everything works fine. And also if after the first run the cloud is un-selected it also works ok (he is probably remembering the last selected object or somethink like that ?)

hope it helps
Luca

**Comment by dgirardeau** [comment]
Yes! Thanks (I couldn't reproduce it on my side).

**Comment by dgirardeau** [comment]
Hum, that's really odd:
- for the polyline tracing tool, I couldn't reproduce the issue. But at least I spotted something strange (the ccDBRoot::addElement method was called directly instead of the standard 'MainWindow::addToDB' one). This can cause errors in some cases.

- but for the interactive segmentation tool, there's no such glitch. And we only export a polyline object in a very particular case (when the user click on the 'export polyline' button) and this is done with the right method (MainWindow::addToDB). It works well (at least on my side).

I can't reproduce this issue, and I really don't see how it can be triggered...

**Comment by luca-penasa** [comment]
for me it seems solved btw. closing it.

**Comment by GinaVulpes** [comment]
I just installed CloudCompare on Ubuntu and I  have just encountered the same issue; when I try to make a selection of points with a polyline I get this message but it is not consistent. Can you please let me know how to solve this?

**Comment by asmaloney** [comment]
Gina:

Would you please create a new issue for this and fill in all the requested info? We need all of the information there (OS version, graphics card, CC version, steps to reproduce, etc.) to help figure out an issue.

Thanks!
